### DISCUSSION TITLE
Vercel not following/finding folder structure when serving static website

### DISCUSSION THREAD

**Discussion by newtonkiragu** [vercel/community#2113] [Help] [answered]
### Summary

I have a static html website that I am deploying using vercel. link can be found [here][1]. However, there are some pages that return `404` yet the directory and index files are present. 

Here is my folder structure.

[![folder structure][2]][2]

As you can see, there is the root folder containing `index.html`. A directory called `sms-solutions` and inside it, a directory called `bulk-and-premium` which contains `index.html`.

The linking of my CSS files in `sms-solutions/index.html` are as follows.
```html
...
<link rel="stylesheet" href="css/style.css" />
...
```
The location of the CSS file can be seen in the folder structure image above, i.e `sms-solutions/css/style.css`.
However, my console shows that Vercel is trying to find the CSS file in the root of the project

My question comes in 3 parts. 

1. Why can I not view the content in the index file when I visit `sms-solutions/bulk-and-premium` or `sms-solutions/bulk-and-premium/index.html`? (I see a 404 Page not found error)
2. Why can't vercel find my static (CSS) files?
3. How can I be able to fix above mentioned issues?

  [1]: https://qlick-website-5ed0y11v0-karanunewton.vercel.app/
  [2]: https://i.stack.imgur.com/p5uRi.png

### Additional information

Question can be found on S.O [here](https://stackoverflow.com/questions/76043139/vercel-not-following-finding-folder-structure-when-serving-static-website)

### Example

https://qlick-website-5ed0y11v0-karanunewton.vercel.app/

**Comment by amyegan** [comment]
Thanks for reaching out, @newtonkiragu. I am not able to replicate the problem on my own static test site. Do you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share here?

**Reply by newtonkiragu** [reply]
Thanks for the prompt reply, @amyegan . [Linked](https://github.com/newtonkiragu/didactic-barnacle) is the public repo for your perusal.

**Reply by amyegan** [reply, marked as answer]
Excellent example! I see the problem now. The nested pages were setting the links relative to the root instead of their parent folder. You can solve this by adding a `vercel.json` config with the contents
```json
{
  "trailingSlash": true
}
```

The [trailingSlash](https://vercel.com/docs/concepts/projects/project-configuration#trailingslash) config will ensure that the path always has a trailing slash to signal that you are in a folder so the relative link references will work
